# microsatABC-IM
Script in R to perform an Approximate Bayesian Computation (ABC) analysis under the Isolation with Migration model from microsatellite data.

Demographic history was inferred from microsatellite data using an approximate Bayesian computation (ABC) approach via random forests (Marin et al. 2016; Pudlo et al. 2016). In this approach, data is simulated from the demographic model with parameter values taken from prior probability distributions and data is transformed into summary statistics. Random forests are used to learn about the parameters from the simulated summary statistics. The resulting random forests can then be used to estimate the posterior probability distributions of parameters.

In this analysis, a model of two populations (1 and 1) is evaluated. Each population is characterized by their present scaled population size θ (θ1=4N1μ and θ2=4N2μ, where N1 is the effective population size of the western population, N2 is the effective population size of the eastern population, and μ is the mutation rate). Western population is founded by individuals from the eastern population at time TF=4N1tF (time tF measured in number of generations) with a founder size θF=4NFμ which changes to θ1 at time T1=4N1t1 (the model assumes an expansion, i.e. θ1>θF). Demographic change is also allowed in the eastern population that has an ancestral size θA=4NAμ that changes to the current size θ2 at time T2=4N1t2. Two concurrent models are evaluated regarding the presence or absence of gene flow between the two population. In the case of presence of gene flow there is an additional parameter, the scaled migration rate M=4N1m. Microsatellites are assumed to mutate following a generalized stepwise mutation model, in which the number of repeat units gained or lost in each mutation is taken from a geometric distribution with parameter PGSM. Data under this model is generated by simulation using coalescent simulator ms (Hudson 2002)

Data input: file in STRUCTURE format (.str) with alleles coded in number of repeats (microsatellite data), two rows per individual, second column indicating the population, missing data coded as -9

### REFERENCES

Hudson, R. R. 2002. Generating samples under a Wright-Fisher neutral model of genetic variation. Bioinformatics 18:337–338.

Marin, J.-M., L. Raynal, P. Pudlo, M. Ribatet, and C. P. Robert. 2016. ABC random forests for Bayesian parameter inference. arXiv:1605.05537.

Pudlo, P., J.-M. Marin, A. Estoup, J.-M. Cornuet, M. Gautier, and C. P. Robert. 2016. Reliable ABC model choice via random forests. *Bioinformatics* **32**:859–866.


### Requirements

Executable of ms and script to read ms output (distributed woth ms) in /bin

R packages: plyr, pegas, mmod, hexbin, grid, abcrf, quantregForest


